src/collection/commands/find.ts

```diff
@@ -33,17 +33,26 @@ export class FindCursor<T> extends CommandCursor<T> {
   }
 
   limit(limit: number): this {
-    this.#context.options.limit = limit;
+    this.#context.options = {
```

Comment:
Why not clone the `context` object at constructor instead?

Reply:
In this case the added unit test will fail, because the options object will have the limit value equals to 10.

Are you ok with updating the `options` parameter?

Reply:
This could be the fix:

```typescript
  constructor(context: FindCursorContext) {
    super(context.protocol, () => this.executor());
    this.#context = {
      ...context,
      options: {
        ...context.options
      }
    };
  }
```

Let me know if you prefer it over the original changes.

Reply:
Last but not least, we can move the fix to the `find` method by changing:

```typescript
  find(
    filter?: Filter<T>,
    options?: FindOptions,
  ): FindCursor<T> {
    return new FindCursor<T>({
      filter,
      protocol: this.#protocol,
      collectionName: this.name,
      dbName: this.#dbName,
      options: options ?? {},
    });
  }
```

this line `options: options ?? {}` with this one: `options: options ? {...options} : {},`.

Reply:
```
constructor(context: FindCursorContext) {
  super(context.protocol, () => this.executor());
  this.#context = {
    ...context,
    options: {
      ...context.options
    }
  };
}
```
This looks good to me